### ISSUE TITLE
TCN parameters

### ISSUE THREAD

**Issue by patnaik42** [abedicodes/ResNet-TCN#6]
Hi,

In addition to above mentioned weighted sampling, I have tried with the below parameters as mentioned in the paper to get 63.9 accuracy with DAiSEE dataset. But training itself is not converging and training accuracy doesn't go beyond 0.5 even till 100 epochs. Can you please confirm if I am missing setting any other parameters?

L = 50 # Number of frames at down sampled rate
Batch-size = 5

ResTCN module parameters:

self.spatial_feat_dim = 512
self.num_classes = 4
self.nhid = 128
self.levels = 8
self.kernel_size = 7
self.dropout = .25

**Comment by patnaik42** [comment]
> Hi,
> 
> In addition to above mentioned weighted sampling, I have tried with the below parameters as mentioned in the paper to get 63.9 accuracy with DAiSEE dataset. But training itself is not converging and training accuracy doesn't go beyond 0.5 even till 100 epochs. Can you please confirm if I am missing setting any other parameters?
> 
> L = 50 # Number of frames at down sampled rate Batch-size = 5
> 
> ResTCN module parameters:
> 
> self.spatial_feat_dim = 512 self.num_classes = 4 self.nhid = 128 self.levels = 8 self.kernel_size = 7 self.dropout = .25



Other information is that, with L = 16, the training converges and I am able to get up to 0.99 training accuracy. Any value of L > 25, training doesn't converge.
Please help me finding what is wrong.

**Comment by abediai** [comment]
Consider the following points in your implementation:
Continuing with L = 16
Train the model with all the samples from the training and validation sets,
During training, use [WeightedRandomSampler](https://pytorch.org/docs/stable/data.html#torch.utils.data.WeightedRandomSampler) in the dataloader to include samples of all classes in all the batches (try larger batch sizes, e.g. 16).
Monitor test accuracy after each training epoch.

**Comment by patnaik42** [comment]
Thank you for the response. I will try your suggestion and check.

**Comment by qq1332427275** [comment]
Hi, have you successfully reproduced the 63.9% accuracy in your paper?


> Thank you for the response. I will try your suggestion and check.

**Comment by qq1332427275** [comment]
@patnaik42 Hi, have you successfully reproduced the 63.9% accuracy ？
